### DISCUSSION TITLE
Qual linguagem Back-end usar?

### DISCUSSION THREAD

**Discussion by dimascamillo** [frontendbr/forum#2072] [Perguntas e Respostas]
Boa noite a todos,

Estou estudando bastante Front-end, e parei para refletir sobre meu plano de estudos e finalmente cheguei na parte de Back-end "voltada para web".

Minha dúvida é:

Qual a linguagem de Back-end devo investir?
Qual linguagem está em alta no mercado?

Quero escolher uma linguagem para me ajudar no desenvolvimento de aplicações web.

Estou no 3º Período da faculdade de Analista de Sistemas.

Estudo e conclui um projeto em Django e tenho conhecimento em PHP puro sem framework.

Desde já agradeço a ajuda!

**Comment by francysreymer** [comment]
Nodejs com typescript.

**Comment by dimascamillo** [comment]
Obrigado pela dica!

**Comment by wilsonneto-swe** [comment]
Quase que qualquer linguagem vai ter muito mercado e estão super aquecidas todas as áreas hoje:

Python + DJango
C# + .Net
Java + Spring
JS/NodeJS + Express ou NestJS
Php + Laravel
....

Olha qual vc mais gosta, brinca com algumas, testa outras.

Eu trabalho com C# + JS/TS + Azure, qualquer uma das que citei acima que escolher está bem servido. O javascript te ajuda no sentido de que você pode trabalhar tanto no backend quanto no frontend com ela, mas no fim não faz tanta diferença assim, a maior parte do tempo eu estou no backend com C# e no front com React.

O mercado de desenvolvimento web está em alta em todas as pontas. Se você já tem conhecimento em Python e em PHP, se gostar de alguma delas investe nelas mesmo ou em apenas uma, python está em alta tanto na parte web quanto na parte de analise de dados e data science, PHP galera vai falar mal até mas tem vagas de monte e continua a cada dia uma linguagem mais robusta e melhor que bate de frente com qualquer outra, NodeJS uma galera vai chamar de Hype e cada dia cresce mais, C# ou Java uma galera vai falar que é coisa antiga mas a cada dia os devs destas tecnologias estão sendo mais procurados e mais bem remunerados, e C# e Java são tão modernas quanto JS/TS...

Enfim, sempre vai ter gente falando que uma linguagem é melhor, que outra é ruim, mas isso é só reflexo das bolhas onde estas pessoas estão, no fim quase todas são ótimas opções e estão super aquecidas.

Bons estudos e sucesso na carreira ai!

**Reply by dimascamillo** [reply]
Nossa que resposta!
Obrigado me esclareceu e motivou muito!
Adaptei meu plano de estudos todinho depois desta resposta!

Novamente obrigado!

**Reply by jpferreiradev** [reply]
Uma das melhores dicas que eu já vi por aqui, valeu mesmo @wilsonneto-dev, principalmente hoje por causa das bolhas e tretas na própria aréa, não sou muito fã sobre as tretas, mas é normal acontecer em qualquer área...

**Reply by ninetails** [reply]
Faça seu plano de estudo levando em conta não só preferências pessoais, mas também empresas onde você quer tentar entrar ou já estiver.

Como o Wilson disse, mercado tá aquecido e tem vaga pra tudo.

Também depende do que vc quer de carreira, já decidiu se será mais frontend, mais backend ou fullstack? Se for o primeiro, creio que bastaria saber e entender de tudo um pouco mas não seria necessário se aprofundar tanto pq será algo que vc pegará durante o trabalho. Já se for pro segundo ou terceiro, aí muda e as escolhas impactarão em escolher uma das opções pra pelo menos começar a aprofundar.
